src/priority/priority.py

```diff
@@ -323,6 +332,26 @@ def insert_stream(self,
         parent.add_child(stream)
         self._streams[stream_id] = stream
 
+    def _stream_cycle(self, new_parent, current):
```

Comment:
This should be hoisted to module scope: a method that never uses the word `self` inside it is just a function with delusions of side effects. 😉